internal/api/handlers.go

```diff
@@ -0,0 +1,172 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+// NewGoCertRouter takes in an environment struct, passes it along to any handlers that will need
+// access to it, then builds and returns it for a server to consume
+func NewGoCertRouter(env *Environment) http.Handler {
+	router := http.NewServeMux()
+	router.HandleFunc("GET /certificate_requests", GetCertificateRequests(env))
+	router.HandleFunc("POST /certificate_requests", PostCertificateRequest(env))
+	router.HandleFunc("GET /certificate_requests/{id}", GetCertificateRequest(env))
+	router.HandleFunc("DELETE /certificate_requests/{id}", DeleteCertificateRequest(env))
+	router.HandleFunc("POST /certificate_requests/{id}/certificate", PostCertificate(env))
```

Comment:
It looks like we are missing the corresponding DELETE to `/certificate_requests/{req-id}/certificate/{cert-id}`

Reply:
hmm, I copied over the methods from the spec. I will add this to the spec and then make the corresponding request available.

Reply:
added the feature to certdb and handlers, and added tests for everything new.